packages/components/src/base-control/index.js

```diff
@@ -7,7 +7,8 @@ function BaseControl( { id, label, help, className, children } ) {
 	return (
 		<div className={ classnames( 'components-base-control', className ) }>
 			<div className="components-base-control__field">
-				{ label && <label className="components-base-control__label" htmlFor={ id }>{ label }</label> }
+				{ label && id && <label className="components-base-control__label" htmlFor={ id }>{ label }</label> }
+				{ label && ! id && <span className="components-base-control__label">{ label }</span> }
```

Comment:
The use-case does not feel legitimate. I don't think we should support `BaseControl` with a `label` and not an `id`. The label should always direct to an input, even in the case of the font size picker.

This leaves open the possibility for error like described at https://github.com/WordPress/gutenberg/pull/10925/files#r257248594 .